### DISCUSSION TITLE
openvpn vs openvpn3 for macOS

### DISCUSSION THREAD

**Discussion by akodithy** [OpenVPN/openvpn3#312] [Q&A]
We are working on providing client VPN solution compatible with OpenVPN for macOS (our VPN server is OpenVPN based) and we are evaluating two options to integrate with client program - openvpn (https://github.com/OpenVPN/openvpn) and openvpn3 (https://github.com/OpenVPN/openvpn3)
 
It seems we can go ahead with openvpn3 as this is newer and more modular version.  However, we want to understand whether there are any cons with using openvpn3 compared to openvpn (like performance, security etc).  Also are there any pros using openvpn compared to openvpn3?

**Comment by schwabe** [comment]
License situation is pretty different (at least at the moment). OpenVPN can be run in a separate process and controlled by management and is under GPL 2.0. OpenVPN3 on the other hand is AGPL3 that is also a library so using OpenVPN3 will make your client solution also AGPL3.

**Reply by akodithy** [reply]
Thanks schwabe.  Assuming OpenVPN3 is run as separate process and we handle IPC between client solution and OpenVPN3 process because of licensing requirements, will there be any other benefits of using OpenVPN3 over OpenVPN?  For example, is OpenVPN3 more secure compared to OpenVPN?  Does OpenVPN3 perform better compared to OpenVPN assuming same client machine and connected to same OpenVPN server?

**Reply by schwabe** [reply]
OpenVPN3 is not designed to be run as a separate process. OpenVPN3 based application will have to be AGPL3 licensed. OpenVPN3 was mainly designed to allow a tight integration with the client. It is programmed in C++ instead C. There is nothing fundamentally more secure or faster in OpenVPN 3 vs OpenVPN 2.

**Comment by Gidro76** [comment]
Good afternoon. When initializing the Openvpn connect connection, the client on MAC OS m1 sonoma version 14.6.1 receives the error: "Error calling protect() method on socket: 30 times". You have to manually launch /Library/Frameworks/OpenVPN Connect.framework/Versions/Current/usr/sbin/vpn agent after which the connection goes through. If you reinstall Openvpn connect, it solves the problem before the first reboot. After the reboot, the error repeats: "Error calling protect() method on socket: 30 times"
